app/services/user_invite_service.rb

```diff
@@ -0,0 +1,26 @@
+class UserInviteService
+  include ServicePattern
+
+  def initialize(user, school, sign_in_url)
+    @user = user
+    @school = school
+    @sign_in_url = sign_in_url
+  end
+
+  attr_reader :user, :school, :sign_in_url
+
+  def call
+    attach_user_to_school
```

Comment:
Sorry to be so anal about this. I think we should do the creation of the user in here, with the `attach_user_to_school` within an [ActiveRecord Transaction block](https://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html) so that in case any of the records fail to save we roll back. I just remembered the active transaction bit, sorry I did not include it in my first review

Something like 

```
def call
  if create user
    send_email
    true
  else
    false
  end
end

def create_user
  ActiveRecord::Base.transaction do
    user.create
    user.attach_to_school
  end
end
```